p2p/proto/snapshot.proto

```diff
@@ -0,0 +1,82 @@
+syntax = "proto3";
+
+import "p2p/proto/common.proto";
+import "p2p/proto/state.proto";
+
+message PatriciaNode {
+    message Edge {
+        uint32 length = 1;
+        Felt252 path = 2; // as bits of left/right
+        Felt252 value = 3;
+    }
+    message Binary {
+        Felt252 left = 1;
+        Felt252 right = 2;
+    }
+
+    oneof node {
+        Edge edge = 1;
+        Binary binary = 2;
+    }
+}
+
+// non leaf nodes required to build the trie given the range (leaves)
+message PatriciaRangeProof {
+    repeated PatriciaNode nodes = 1;
+}
+
+// leafs of the contract state tre
+message ContractState {
+    Hash address = 1; // the key
+    Hash class = 2;
+    Hash storage = 3; // patricia
+    uint64 nonce = 4;
+}
+
+// request a range from the contract state tree that matches the given root (block)
+// starts at 'start' and ends no less than 'end'.
+// the result is  (ContractRange+, PatriciaRangeProof)*
+message GetContractRange {
+    Hash root = 2;
+    Hash start = 3;
+    Hash end = 4;
+    uint32 chunks_per_proof = 5; // how many ContractRange items to send before sending a proof
```

Comment:
You mean, `max_leafs`? or `max_nodes`?

Reply:
Sorry, not sure what you're referring to.

Reply:
Basically, I don't understand what this is. With ethereum snap, there is a maximum size (total of all contract) for the response. eg: https://github.com/NethermindEth/juno/pull/1030/files#diff-a04fd2fd0785454e284c95dea53642c1d266c8cbbf66a8862b330362cf710e87R76. As most contract are small, all the leafs can be transferred in one request, so no proof is needed. The exception is the last trie, which is not complete because the limit is exceeded during that trie, so the last trie need the proof.

Reply:
The thought was that the client may want a proof more or less frequent according to their memory or how often it wants to validate the chunks.

Reply:
Right, I'm guessing the intent it to stream the leafs?

Reply:
Yes